realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp

```diff
@@ -85,10 +85,19 @@ class JniConfigWrapper {
         auto error_handler = [=](std::shared_ptr<SyncSession> session, SyncError error) {
             realm::jni_util::Log::d("error_handler lambda invoked");
 
-            JNIEnv* env = realm::jni_util::JniUtils::get_env(true);
+            auto error_message = error.message;
+            auto error_code = error.error_code.value();
+            if (error.is_client_reset_requested()) {
+                // Hack the error message to send information about the location of the backup.
+                // If more uses of the user_info map surfaces. Refactor this to send the full
+                // map instead.
+                error_message = error.user_info[SyncError::c_recovery_file_path_key];
+                error_code = 7; // See ErrorCode.java
```

Comment:
I'm not sure it is done out of the box, but otherwise we can probably make it happen. It would be much safer than this at least